pybossa/cache/users.py

```diff
@@ -518,12 +519,16 @@ def get_announcements_cached(user, announcement_levels):
 
 def get_users_for_data_access(data_access):
     from pybossa.data_access import get_user_data_access_db_clause
-
     clause = get_user_data_access_db_clause(data_access)
     if not clause:
         return None
 
-    sql = text('''select id::text, fullname from "user" where {}'''.format(clause))
+    default_user_access_levels = app_settings.config.get('DEFAULT_USER_ACCESS_LEVELS')
+    if default_user_access_levels and bool(set(data_access).intersection(set(default_user_access_levels))):
+        sql = text('''select id::text, fullname from "user"''')
+    else:
+        sql = text('''select id::text, fullname from "user" where {}'''.format(clause))
+
```

Comment:
User profile page not getting updated to pick either L3 or L4 which means all users by default gets L3, L4 access on public. So listing all users to be selected in "Assign users" on public. And only added users to the project can contribute it. No need to check against data access will  make it simple i think and solve the purpose.